### PR TITLE
[TT-5094] Implement validate request middleware in OAS

### PR DESCRIPTION
**Pull request by furkansenharputlu**

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4007/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2167316797)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4007/merge
Commit: fdea1ca8be69e730e2ee75d4e8faacbdacf473fb 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2167372138)

**Review by titpetric — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4007/merge
Commit: 99e0b4628ebe00c83fc0e6a981d2a703695f0fdb 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2172838697)

File: apidef/oas/operation.go [hunk 1]
```diff
@@ -343,3 +347,144 @@ func (x *XTykAPIGateway) getOperation(operationID string) *Operation {
...
+			operation.RequestBody = requestBody
+		}
+
+		bodyContent := requestBody.Value.Content
+		if bodyContent == nil {
+			bodyContent = openapi3.NewContent()
+			requestBody.Value.Content = bodyContent
+		}
+
+		mediaType := bodyContent.Get("application/json")
```
Comment (jeffy-mathew) [inline apidef/oas/operation.go:411]: shall we define a const for `application/json` and use it  instead of string literal?

**Review by jeffy-mathew — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4007/merge
Commit: 06cab8b3b06cc93c330e388becf25491f2b752c6 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2183819467)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4007/merge
Commit: 6d34889f12640cefd0d219f0b839035b73cd7200 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2183853763)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4007/merge
Commit: 519eb4d5062f8ca3c72365e4d05e1b04e6f08ecd 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2183916103)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 6abaf1376db86e85da148483b9edd52f18277619 [TT-5094] Implement validate request middleware in OAS (#4007)

* Implement validate request middleware in OAS

* Log errors instead of ignoring

* Add constant contentTypeJSON
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2184191727)

**Pull request merged**